addon/templates/components/columns/base.hbs

```diff
@@ -2,7 +2,9 @@
   {{component column.component column=column table=table tableActions=tableActions extra=extra sortIcons=sortIcons}}
 {{else}}
   {{label}}
-  {{#if sortIconProperty}}
+  {{#if (and sortIcons.iconComponent sortIconProperty)}}
+    {{component sortIcons.iconComponent sortIcons=sortIcons sortIconProperty=sortIconProperty}}
```

Comment:
would you mind adding a test to verify the arguments (`sortIcons=sortIcons sortIconProperty=sortIconProperty`) get passed to the component declared by  `iconComponent` when we render a table like you do in the demo app:

```diff
+  {{t.head
+    onColumnClick=(action 'onColumnClick')
+    iconSortable='unfold_more'
+    iconAscending='keyboard_arrow_up'
+    iconDescending='keyboard_arrow_down'
+    iconComponent='materialize-icon'
+    fixed=true
+  }}
```

You can steal some ideas from
https://github.com/cibernox/ember-power-select/blob/master/tests/integration/components/power-select/customization-with-components-test.js#L257..L281